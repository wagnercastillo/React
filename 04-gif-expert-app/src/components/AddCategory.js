import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategorias }) => {
	const [inputValue, setInputValue] = useState("");

	const handleInputChange = (e) => {
		setInputValue(e.target.value);
	};

	const handleSubmit = (e) => {
		e.preventDefault();
		
		if (inputValue.trim().length > 2) {
			setCategorias((categorias) => [inputValue, ...categorias]);
			setInputValue('');
		}
	};
	return (
		<form onSubmit={handleSubmit}>
			<p>{inputValue}</p>
			<input 
				type="text"
				value={inputValue}
				onChange={handleInputChange} 
			/>
		</form>
	);
};

AddCategory.defaultProps = {
	setCategorias: { setCategorias: ["One Punch", "Samurai X", "Dragon Ball"] },
};

AddCategory.propTypes = {
	setCategorias: PropTypes.func.isRequired,
};
