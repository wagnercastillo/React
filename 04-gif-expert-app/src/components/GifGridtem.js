import React from "react";
import PropTypes from "prop-types";

export const GifGridtem = ({ id, title, url }) => {
	return (
		<div className="card animate__animated animate__bounce animate__delay-2s">
			<img src={url} alt={title} />
			<p>{title}</p>
		</div>
	);
};

GifGridtem.propTypes = {
	title: PropTypes.string.isRequired,
	url: PropTypes.string.isRequired,
};
