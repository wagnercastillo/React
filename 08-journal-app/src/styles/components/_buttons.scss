.btn {
    background-color: transparent;
    border: none;
    color: white;
    cursor: pointer;
    font-size: 12px;
    padding: 7px 10px 7px 10px;

    &:focus {
        outline: none;
    }

    &:hover {
        color: darken($color: $primary, $amount: 15);
    }
}

.btn-primary {
    background-color: $primary;
    border-radius: 2px;


    &:disabled {
        background-color: lighten($color: $primary, $amount: 15);
    }

}

.btn-block{
    width: 100%;
}

.google-btn {
    cursor: pointer;
    margin-top: 5px;
    background-color: #4285f4;
    box-shadow: 0 3px 4px 0 rgba(0, 0, 0, 0.25);
    transition: box-shadow .3s ease;
    height: 42px;
    display: flex;
    align-items: center;
    justify-content: space-between;
}

.google-icon-wrapper {
    background-color: $white;
    height: 42px;
    width: 42px;
    display: flex;
    align-items: center;
    justify-content: center;
}

.google-icon {
    width: 22px;
}

.btn-text {
    padding: 0 14px;
    color: $white;
    font-size: 14px;
    letter-spacing: 0.2px;

    &:hover {
        box-shadow: 0 0 6px $google-blue
    }

    &:active {
        background: $button-active-blue
    }
}