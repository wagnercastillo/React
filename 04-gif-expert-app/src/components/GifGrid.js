import React from "react";
import PropTypes from "prop-types";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridtem } from "./GifGridtem";

export const GifGrid = ({ categoria }) => {
	const { data: images, loading } = useFetchGifs(categoria);

	return (
		<>
			<h3 className="animate__animated animate_fadeIn">{categoria}</h3>

			{loading && <p className="animate__animated animate__fadeIn">Loading</p>}

			<div className="card-grid">
				{images.map((img) => (
					<GifGridtem key={img.id} {...img} />
				))}
			</div>
		</>
	);			
};

GifGrid.propTypes = {

	categoria: PropTypes.string.isRequired, 

}