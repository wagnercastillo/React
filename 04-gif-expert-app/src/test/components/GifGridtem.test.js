import React from "react";
import { shallow } from "enzyme";
import { GifGridtem } from "../../components/GifGridtem";

describe("Pruebas en <GifGrid />", () => {

	const title = "Titulo de prueba";
	const url = "https://localhost/algo.jpg";

	const wrapper = shallow(<GifGridtem title={title} url={url} />);

	test("Debe de mostrar el componente correctamente", () => {
		expect(wrapper).toMatchSnapshot();
	});

	test("Debe de tener un parrafo con title", () => {
		const p = wrapper.find("p");
		expect(p.text().trim()).toBe(title);
	});

	test("Debe de tener la imagen igual al url y alt de los props", () => {
		const img = wrapper.find("img");
		expect(img.prop("src")).toBe(url);
		expect(img.prop("alt")).toBe(title);
	});

	test("Debe de tener animate__fadeIn", () => {
		const div = wrapper.find("div");
		expect(div.hasClass("animate__animated")).toBe(true);

	});
});
