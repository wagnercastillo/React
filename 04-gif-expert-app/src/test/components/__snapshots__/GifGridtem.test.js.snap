// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__bounce animate__delay-2s"
>
  <img
    alt="Titulo de prueba"
    src="https://localhost/algo.jpg"
  />
  <p>
    Titulo de prueba
  </p>
</div>
`;
